### PR TITLE
Fake filesystem changes to devfile parser

### LINKED ISSUE #2553: PR #2500 Followup
- [x] https://github.com/openshift/odo/pull/2500#discussion_r372890591 let's use the fake fs package here and write the stuff in memory instead of the disk. That will speed up the testing.
- [x] https://github.com/openshift/odo/pull/2500#discussion_r372891177 why not use the Fs Filesystem which is part of the DevfileCtx? What will help us with unit tests using the fake system.
- [x] https://github.com/openshift/odo/pull/2500#discussion_r372891839 We can pass the fake filesystem here in DevfileCtx
- [ ]  https://github.com/openshift/odo/pull/2500#discussion_r372895747 since the description says run command not present, should this be build app?
- [x] https://github.com/openshift/odo/pull/2500#discussion_r372906855 can we convert these tests to the tabular format we currently use in odo?

/assign @kanchwala-yusuf 
/kind devfile
/label state/Ready

### PR DESCRIPTION
**Pull request by kanchwala-yusuf**
**What type of PR is this?**
> /kind feature

**What does does this PR do / why we need it**:
Accommodating review comments from #2500 

**Which issue(s) this PR fixes**:

Fixes #2553 

**How to test changes / Special notes to the reviewer**:
Devfile related actions should work without any errors.

### CODE REVIEW

File: pkg/devfile/parser/content_test.go [hunk 1]
```diff
@@ -39,8 +40,12 @@ func TestSetDevfileContent(t *testing.T) {
 	t.Run("valid file", func(t *testing.T) {
 
 		var (
-			tempDevfile = createTempDevfile(t, validJsonRawContent100())
-			d           = DevfileCtx{absPath: tempDevfile.Name()}
+			fakeFs      = filesystem.NewFakeFs()
```
Comment (mik-dass) [inline pkg/devfile/parser/content_test.go]: shouldn't this test be converted to table format?

Comment (girishramnani) [reply pkg/devfile/parser/content_test.go]: +1 on this

**Pull request merged**